src/gpuarray/buffer.h

```diff
@@ -90,6 +90,19 @@ GPUARRAY_PUBLIC gpucontext *gpucontext_init(const char *name, int dev,
 #define GA_CTX_SINGLE_THREAD 0x02
 
 /**
+ * Perform all operations in the order they appear.
+ *
+ * This will remove any attempt at exploiting parallelism in the
+ * underlying device by performing unrelated operations concurrently
+ * and/or out of order.
+ *
+ * This can help performance by removing the small cost paid for each
+ * operation to keep everything coherent in the face of parallelism.
+ * It can also hinder performance by doing everything one after the other.
+ */
+#define GA_CTX_SINGLE_STREAM 0x4
```

Comment:
Can you at least tell that this is one stream per context? Like in the first line:

 * Use only 1 steram per context, so this force to perform all operations in the order they appear for one context.